### DISCUSSION TITLE
Override for normal key bindings in Telega?

### DISCUSSION THREAD

**Discussion by hochata** [meow-edit/meow#280] [Q&A] [answered]
Hi! I'm just trying out Meow and so far I find it very awesome!

But I have a doubt. It seems that the Telega keybindings in `telega-chat-mode` override the normal keyebindings. In particular `SPC` (so no Keypad mode access) and `k` (so no upward navigation).

Is there a way to rewrite the commands like `meow-motion-overwrite-define-key` does but for normal keybindings?

**Comment by eshrh** [comment]
Is the issue that meow doesn't enter motion mode in telega? If so could you try adding `telega-mode` (or whatever the mod is called (call C-h m)) to the list `meow-mode-state-list` and see if that fixes it?

**Reply by hochata** [reply]
I thought that might help but it doesn't. Even in motion mode, the bindings set by Telega take precendece for some reason. 

For now I just manually rewrote the problematic keybindings.

**Reply by hochata** [reply]
After looking a little bit deeper, I think the Telega keybindings take priority because they are associated with buttons. So while hovering over the button, the button keybindings overwrite Meow's keybindings.

**Reply by DogLooksGood** [reply]
Telega attach keybinds with overlays, if I understand correctly. It's already the highest priority, so Meow can't give itself a higher priority.
I just remove the `SPC` in chat buffer.
```
(define-key telega-msg-button-map (kbd "SPC") nil)
```

**Reply by hochata** [reply]
Thanks for the info! That worked like a charm!

**Comment by hochata** [comment, marked as answer]
For completeness sake, I'll copy @DogLooksGood answer here.

> Telega attach keybinds with overlays, if I understand correctly. It's already the highest priority, so Meow can't give itself a higher priority. I just remove the `SPC` in chat buffer.
> 
> ```
> (define-key telega-msg-button-map (kbd "SPC") nil)
> ```

This solves the problem!
